smithy-aws-protocol-tests/model/awsJson1_0/unions.smithy

```diff
@@ -43,6 +43,7 @@ union MyUnion {
     listValue: StringList,
     mapValue: StringMap,
     structureValue: GreetingStruct,
+    unitValue: Unit,
```

Comment:
Can the `Unit` target entry be separated out to its own test case?

Reply:
removed—I thought it was just an oversight that it was missing. Turns out this actually causes compilation failure in the Rust server codegen: https://github.com/smithy-lang/smithy-rs/issues/2546

So we should definitely get a test with this at some point but it will probably break a few folks